### DISCUSSION TITLE
add_mesh_clip_plane - can i hide the bounding box and plane when actor.SetVisibility(False)

### DISCUSSION THREAD

**Discussion by JohnForeman** [pyvista/pyvista#4195] [Q&A] [answered]
Hi all,
i'm using a hot key to switch between a mesh that is clipped and one that is not.  that is working fine but when i use the actor.SetVisibility(False) for the clipped mesh the bounding box and plane are still visible.  is it possible to toggle the visibility of the bound box and plane also?
thanks

**Comment by banesullivan** [comment, marked as answer]
You'll want to use the `.SetEnabled()` API of the widget object itself. You can access the box widget under the `Plotter.box_widgets` attribute. Here is an example to use the `b` key to toggle the widget on and off:
```py
import pyvista as pv

pl = pv.Plotter(notebook=0)
pl.add_mesh_clip_box(pv.Wavelet())
widget = pl.box_widgets[0]
def toggle(*args):
    widget.SetEnabled(not widget.GetEnabled())
pl.add_key_event('b', toggle)
pl.show()
```

**Reply by banesullivan** [reply]
It might be nice if we had a button to toggle all of the widgets like this easily by default. I would say to use `w` but that toggles to wireframe representation style

**Reply by JohnForeman** [reply]
i bumped into the built in 'w' and 's' hotkey that caused me a little pain before i realized the were switching between wireframe and shading.  is there a list of built in hot keys?  so i don't add to them unknowingly

**Reply by banesullivan** [reply]
I think all are listed at the top of this page: https://docs.pyvista.org/api/plotting/plotting.html

Further, in code, the are configured here: 

https://github.com/pyvista/pyvista/blob/344f8388841043e0ff68b5813c4560f55333f057/pyvista/plotting/plotting.py#L1982-L1999

FYI: https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.clear_events_for_key.html

**Reply by JohnForeman** [reply]
sorry.  i didn't notice that info.  thanks

**Comment by JohnForeman** [comment]
that worked.  thanks
